### DISCUSSION TITLE
Library/Dependency information

### DISCUSSION THREAD

**Discussion by dphaldes** [build-server-protocol/build-server-protocol#666] [Proposals]
This is a continuation of https://github.com/microsoft/language-server-protocol/issues/472. The thread points out that this could/should be the responsibility of the BSP and not the language itself. 
This feature is really useful and language/toolchain agnostic way of getting information of dependencies would be really useful. Things like Code Examples, Usage documentation and Implementation details help a lot when using libraries (it's commonly used in Java with embedded docs.)

**Comment by tgodzik** [comment]
I think this is already done via https://build-server-protocol.github.io/docs/specification#buildtargetdependencymodules-request 

It's mostly currently used for Maven dependencies, you can get jars, source jars or javadocs for each build target using that endpoint. Would that be enough for you? We have examples of it in bazel BSP, Bloop and from the client side in Metals language server.

**Reply by dphaldes** [reply]
Does it fetch just the dependency names, or also their files and its locations ?

**Reply by tgodzik** [reply]
All of that: https://build-server-protocol.github.io/docs/extensions/maven#mavendependencymodule plus version and org
